### PR TITLE
don't depend directly on micronaut validation

### PR DESCRIPTION
**Pull request by sdelamo**
Close #217 #216 

This PR changes so that we depend only `jakarta-validation-api`.

### CODE REVIEW

File: multitenancy/build.gradle.kts [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
     annotationProcessor(mnValidation.micronaut.validation.processor)
-    implementation(mnValidation.micronaut.validation)
-
+    api(mnValidation.validation)
     api(mn.micronaut.http)
     api(mn.micronaut.inject)
 
     compileOnly(mnSession.micronaut.session)
 
+    testImplementation(mnValidation.micronaut.validation)
```
Comment (sdelamo) [inline multitenancy/build.gradle.kts:13]: If I change this from `testImplementation` to `implementation` the test annotated with `@PendingFeature` in this PR passes. 

@dstepanov do you understand why it does not pass with micronaut-validation on the test classpath?

Comment (dstepanov) [reply multitenancy/build.gradle.kts:13]: It is strange

File: multitenancy/src/main/java/io/micronaut/multitenancy/tenantresolver/HttpRequestTenantResolver.java [hunk 2]
```diff
@@ -36,5 +37,5 @@ public interface HttpRequestTenantResolver {
      * @throws TenantNotFoundException if tenant not found
      */
     @NonNull
-    Serializable resolveTenantIdentifier(@NonNull HttpRequest<?> request) throws TenantNotFoundException;
+    Serializable resolveTenantIdentifier(@NonNull @NotNull HttpRequest<?> request) throws TenantNotFoundException;
```
Comment (dstepanov) [inline multitenancy/src/main/java/io/micronaut/multitenancy/tenantresolver/HttpRequestTenantResolver.java:40]: Do we want to add validation annotations? IMHO, internal code should limit the usage as it adds to the performance. In this case, we can do manual checks.

Comment (sdelamo) [reply multitenancy/src/main/java/io/micronaut/multitenancy/tenantresolver/HttpRequestTenantResolver.java]: I have removed validation and added `Objects.nonNull(request, "request must not be null)` inside the methods. 

https://github.com/micronaut-projects/micronaut-multitenancy/pull/218/commits/20251aa0f489d19964a058ff26381c04cc3469a7

Comment (sdelamo) [reply multitenancy/src/main/java/io/micronaut/multitenancy/tenantresolver/HttpRequestTenantResolver.java]: ping @dstepanov

**Review by dstepanov — APPROVED** [review]

**Pull request merged**